### PR TITLE
Text In mk2: sort output sockets, if order is not defined.

### PR DESCRIPTION
**Pull request by portnov**
## Addressed problem description

#2126 

## Solution description

Order outputs by using sorted(), if "sockets_order" is not defined.

## Preflight checklist

Put an x letter in each brackets when you're done this item:

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [ ] Unit-tests implemented.
- [x] Ready for merge.

### CODE REVIEW

**Comment by portnov** [comment]
@zeffii please review.

**Comment by zeffii** [comment]
whether you sort the sockets or not, it won't really fix all nodes. it might fix some older jsons ,,  like like a broken clock will give the correct time 2 times a day.

**Comment by zeffii** [comment]
whether you sort the sockets or not, it won't really fix all older jsons (sans socket_order). it might fix some older jsons ,, just like a broken clock will give the correct time 2 times a day.

**Comment by zeffii** [comment]
i'm all for including a fix for older nodes, but realistically all old jsons containing Text In node were fundamentally broken. If they worked, it seems they were accidentally ordered as expected.

**Comment by zeffii** [comment]
!!!! oh I know why this is only now coming to the forefront!

 initially socket where linked using their socket `.  name`, i changed this socket `. index ` some time ago. This would have broken a few older exports :)

**Comment by portnov** [comment]
Eeh. So, is it okay to merge this?

**Comment by zeffii** [comment]
### Problem
a brief explanation.

I know why this is only now coming to the forefront!

initially io/json stored socket links using their socket `.  name`, i changed this (for good reasons) to socket `. index ` some time ago. This means that whatever the socket order was, didn't matter because links where remade using socket.name . even when the node recreated it's own sockets (however random that was) the relinking was correct.

When i switched to socket.index this `text_in` reloading/relinking was broken. 

### Solution

the proposed solution will sort keys alphabetically, for UI regenerating pre-linking


### My issue with the proposed solution

It will fix some older jsons, but as a node outputs more sockets the chances increase that even `sorted(outputs.items() )` will not place the sockets in the correct order.

### tldr

not sure there's any nice solution for old jsons prior to `socket_order`

**Comment by portnov** [comment]
Huh. Let's look at the problem like so: does this change make things better or worse? As I see it:
* Before this change, order was random, so for example for node with 4 outputs there was about 1/24 chance that it would load correctly. For newly saved jsons, you could not know if it would load or not.
* After this change, order is fixed. For earlier existing jsons, there still is about 1/24 chance that it would load correctly (in case when initial order of sockets was exactly alphabetical). But for newly saved jsons, you know that it would load correctly.

**Comment by zeffii** [comment]
you can of course commit this.  it doesn't affect me.

but, because it will still fail for some gists we try. 

So we can also offer a few steps for users to resolve their gists in the case sorting also didn't help.
```
import bpy


def get_socket_order(n):
    socket_order = [socket.name for socket in n.outputs]
    print('\"socket_order\":', str(socket_order))
        

n = bpy.data.node_groups['NodeTree'].nodes.active
get_socket_order(n)
```
this code could be executed 1 time for each text_in node, just to get something like this:
```python
"socket_order": ['Objects input:Edges', 'Objects input:Matrixes', 'Objects input:Polygons', 'Objects input:Vertices']
```
and you would use that to manually edit define the  `socket_order` , _after_ noticing that the no automatic sorting was producing the expected order.

**Comment by zeffii** [comment]
this way
```diff
"socket_order": ['Objects input:Edges', 'Objects input:Matrixes', 'Objects input:Polygons', 'Objects input:Vertices']
"socket_order": ["Objects input:Edges", "Objects input:Matrixes", "Objects input:Polygons", "Objects input:Vertices"]
```

**Comment by portnov** [comment]
Included in #2108 .
